src/analysis_and_optimization/Partial_evaluator.ml

```diff
@@ -509,6 +510,7 @@ let rec eval_expr (e : Expr.Typed.t) =
             | "pow", [x; {pattern= FunApp (StanLib, "Divide__", [y; z]); _}]
               when is_int 1 y && is_int 2 z ->
                 FunApp (StanLib, "sqrt", [x])
+                (* This is wrong; if both are type UInt the exponent is rounds down to zero. *)
```

Comment:
So if someone does `pow(x, 1/2)` that would technically have to return 1, but we fix that and replace with sqrt(x). I feel what we do is fine, its just not consistent.

Reply:
The result of `pow(x, 1/2)` depends on optimization level. At levels O2 or higher the partial evaluator changes it to `sqrt(x)`. At lower optimization level the partial evaluator does not run and it is up to the backend to decide what to do with `1/2`. C++ rounds it to zero, while Python (TFP) does proper division and gets `0.5`.
I agree this "inconsistency" is fine. That's why I didn't change the code, just added a comment.

Reply:
But `pow(x, 1/2)` will still warn right?

Reply:
Yes, the warning is always there.